### PR TITLE
feat(interaction): refactor action of element and support elementSelected interaction

### PR DESCRIPTION
**Pull request by visiky**
Design: https://www.yuque.com/antv/g2-docs/interaction-plan#eo3HY

### Description

- **refactor(action)**: `ActiveElement` action supports `border` option
- **refactor(action)**: `ElementSelection` supports `multiple` and `toggle` options
- **feat(interaction)**: add `ElementSelected` interaction, which enable `singleMode` option to achieve `ElementSingleSelected` interaction
- **docs**: add examples of Element Interaction

### InteractionDescriptor

- InteractionDescriptor of `ElementSelected`

```ts
export const InteractionDescriptor = (options: ElementSelectedOptions = {}) => {
  const { singleMode = false } = options;
  return {
    start: [
      {
        trigger: 'plot:pointerdown',
        action: [
          { type: 'elementSelection', multiple: !singleMode, toggle: true },
          { type: 'activeElement', border: 2, ...options },
        ],
      },
    ],
  };
};
```

#### ElementSelected

```ts
G2.render({
  type: 'interval',
  data: [
    { city: 'London', month: 'Jan.', rainfall: 18.9 },
    { city: 'London', month: 'Feb.', rainfall: 28.8 },
    { city: 'London', month: 'Mar.', rainfall: 39.3 },
    { city: 'London', month: 'Apr.', rainfall: 81.4 },
    { city: 'London', month: 'May', rainfall: 47 },
    { city: 'London', month: 'Jun.', rainfall: 20.3 },
    { city: 'London', month: 'Jul.', rainfall: 24 },
    { city: 'London', month: 'Aug.', rainfall: 35.6 },
    { city: 'Berlin', month: 'Jan.', rainfall: 12.4 },
    { city: 'Berlin', month: 'Feb.', rainfall: 23.2 },
    { city: 'Berlin', month: 'Mar.', rainfall: 34.5 },
    { city: 'Berlin', month: 'Apr.', rainfall: 99.7 },
    { city: 'Berlin', month: 'May', rainfall: 52.6 },
    { city: 'Berlin', month: 'Jun.', rainfall: 35.5 },
    { city: 'Berlin', month: 'Jul.', rainfall: 37.4 },
    { city: 'Berlin', month: 'Aug.', rainfall: 42.4 },
  ],
  encode: {
    x: 'month',
    y: 'rainfall',
    color: 'city',
  },
  interaction: [{ type: 'elementSelected' }],
});
```

#### ElementSingleSelected
element-single-selected

```ts
G2.render({
  type: 'interval',
  data: [
    { genre: 'Sports', sold: 275, type: 'A' },
    { genre: 'Strategy', sold: 115, type: 'B' },
    { genre: 'Action', sold: 120, type: 'C' },
    { genre: 'Shooter', sold: 350, type: 'D' },
    { genre: 'Other', sold: 150, type: 'E' },
  ],
  encode: {
    x: 'genre',
    y: 'sold',
    color: 'genre',
  },
  interaction: [{ type: 'elementSelected', singleMode: true }],
});
```

#### ElementSelected custom `color` and `border` options

custom-element-selected

```ts
G2.render({
  type: 'interval',
  data: [
    { genre: 'Sports', sold: 275, type: 'A' },
    { genre: 'Strategy', sold: 115, type: 'B' },
    { genre: 'Action', sold: 120, type: 'C' },
    { genre: 'Shooter', sold: 350, type: 'D' },
    { genre: 'Other', sold: 150, type: 'E' },
  ],
  encode: {
    x: 'genre',
    y: 'sold',
    color: 'genre',
  },
  interaction: [{ type: 'elementSelected', color: 'red', border: 3 }],
});
```

### CODE REVIEW

File: src/interaction/builtin/elementHighlight.ts [hunk 1]
```diff
@@ -7,13 +7,19 @@ export const InteractionDescriptor = (options?: ElementHighlightOptions) => ({
   start: [
     {
       trigger: 'plot:pointermove',
-      action: [{ type: 'elementSelection' }, { type: 'highlight', ...options }],
+      action: [
+        { type: 'elementSelection' },
+        { type: 'highlightElement', ...options },
```
Comment (visiky) [inline src/interaction/builtin/elementHighlight.ts:12]: 上次改动，没有修正过来

File: src/spec/interaction.ts [hunk 2]
```diff
@@ -31,6 +32,13 @@ export type ElementActiveInteraction = {
   color?: string;
 };
 
+export type ElementSelectedInteraction = {
+  type?: 'elementSelected';
+  color?: string;
+  border?: number;
+  singleMode?: boolean;
```
Comment (visiky) [inline src/spec/interaction.ts:39]: 单独一个 `ElementSingleSelected` 交互，还是通过配置项就好？之前是独立开的两个交互

Comment (pearmini) [reply src/spec/interaction.ts:39]: 目前都行，主要是看当前其他 action 是如何区分的。

**Review by pearmini — APPROVED** [review]

File: src/spec/interaction.ts (hunk 2 above)
Comment (visiky) [reply src/spec/interaction.ts:39]: ok，相似的【交互】先统一用配置项区分。后面看其它交互独立拆开，还是合并通过配置项区分，哪个方式更好。

**Pull request merged**